src/provider/qti.js

```diff
@@ -492,18 +492,40 @@ var qtiProvider = {
             })
             .on('pause', function(data) {
                 const testContext = self.getTestContext();
-
+                let baseParams = {
+                    itemDefinition: testContext.itemIdentifier,
+                    reason: {
+                        reasons: data && data.reasons,
+                        comment: data && (data.originalMessage || data.message)
+                    }
+                }
                 this.setState('closedOrSuspended', true);
 
-                this.getProxy()
-                    .callTestAction('pause', {
-                        itemDefinition: testContext.itemIdentifier,
-                        itemState: self.itemRunner.getState(),
-                        reason: {
-                            reasons: data && data.reasons,
-                            comment: data && (data.originalMessage || data.message)
+                // get item response depends on availability item runner
+                const itemState = new Promise((resolve, reject) => {
+                    if(self.itemRunner.isCleared()){
+                        const proxy = this.getProxy();
+                        const store = proxy && proxy.itemStore;
```

Comment:
Do all proxies that can be used with this provider have itemStore - is it some required interface? Since you reject if it doesn't exist

Reply:
Good catch Olga!
Indeed, this API is not standard, and the item store is only available in the context of the pre-caching proxy.

I don't think this should be made this way. The test runner implementation must not tamper with the proxy internals.